scripts/rddepman.ts

```diff
@@ -0,0 +1,210 @@
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = getTitle(name, currentVersion, latestVersion);
+  const branchName = getBranchName(name, currentVersion, latestVersion);
+
+  console.log(`Creating PR "${ title }".`);
+  await getOctokit().rest.pulls.create({
+    owner: GITHUB_OWNER,
+    repo:  GITHUB_REPO,
+    title,
+    base:  MAIN_BRANCH,
+    head:  branchName,
+  });
+}
+
+type PRListFn = ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>;
+
+async function getPulls(name: string): Promise<Awaited<PRListFn>['data']['items']> {
+  const queryString = `type:pr repo:${ GITHUB_OWNER }/${ GITHUB_REPO } head:rddepman/${ name }`;
+  let response: Awaited<PRListFn>;
+  let retries = 0;
+
+  while (true) {
+    try {
+      response = await getOctokit().rest.search.issuesAndPullRequests({ q: queryString });
+      break;
+    } catch (error: any) {
+      retries += 1;
+      if (retries > 2) {
+        throw error;
+      }
+    }
+  }
+
+  return response.data.items;
+}
+
+async function checkDependencies(): Promise<void> {
+  // exit if there are unstaged changes
+  git('update-index', '--refresh');
+  if (git('diff-index', '--quiet', 'HEAD', '--')) {
+    console.log('You have unstaged changes. Commit or stash them to manage dependencies.');
+
+    return;
+  }
+
+  // load current versions of dependencies
+  const depVersionsPath = path.join('src', 'assets', 'dependencies.yaml');
+  const currentVersions = await readDependencyVersions(depVersionsPath);
+
+  // get a list of dependencies' version comparisons
+  const versionComparisons: VersionComparison[] = await Promise.all(dependencies.map(async(dependency) => {
+    return {
+      name:           dependency.name,
+      currentVersion: currentVersions[dependency.name as keyof DependencyVersions],
+      latestVersion:  await dependency.getLatestVersion(),
+    };
+  }));
+
+  // filter comparisons down to the ones that need an update
+  const updatesAvailable = versionComparisons.filter(({ name, currentVersion, latestVersion }) => {
+    const equal = compareVersions(currentVersion, latestVersion);
+
+    if (equal) {
+      console.log(`${ name } is up to date.`);
+    } else {
+      console.log(`Can update ${ name } from ${ currentVersion } to ${ latestVersion }`);
+    }
+
+    return !equal;
+  });
+
+  // reconcile dependencies that need an update with state of repo's PRs
+  const needToCreatePR: VersionComparison[] = [];
+
+  await Promise.all(updatesAvailable.map(async({ name, currentVersion, latestVersion }) => {
+    // try to find PR for this combo of name, current version and latest version
+    const prs = await getPulls(name);
+
+    // we use title, rather than branch name, to filter pull requests
+    // because branch name is not available from the search endpoint
```

Comment:
For my own reference: [listing pulls](https://api.github.com/repos/rancher-sandbox/rancher-desktop/pulls) has a option for `head=`, but that only does exact matches, so it's not appropriate here.

Reply:
Yeah... it's a bummer, since the objects returned from the pulls endpoint have more info about the PR.